spec.html

```diff
@@ -787,68 +787,97 @@ <h1>Throw an Exception</h1>
+          <li>_hygienicTemp_ is ephemeral and visible only in the steps pertaining to ReturnIfAbrupt.
+        </ul>
+
+        <p>For example, an algorithm step that says:</p>
         <emu-alg>
-          1. Let _result_ be AbstractOperation(ReturnIfAbrupt(_argument_)).
+          1. Let _result_ be _someValue_ if ReturnIfAbrupt(_someValue_.OperationName(_firstArgument_, _secondArgument_)) is *true*, and _someOtherValue_ otherwise.
         </emu-alg>
-        <p>mean the same thing as:</p>
+        <p>is equivalent to the sequence of steps:</p>
         <emu-alg>
-          1. If _argument_ is an abrupt completion, return _argument_.
-          1. If _argument_ is a Completion Record, set _argument_ to _argument_.[[Value]].
-          1. Let _result_ be AbstractOperation(_argument_).
+          1. Let _hygienicTemp_ be _someValue_.OperationName(_firstArgument_, _secondArgument_).
+          1. If _hygienicTemp_ is an abrupt completion, return _hygienicTemp_.
+          1. Else if _hygienicTemp_ is a Completion Record, set _hygienicTemp_ to _hygienicTemp_.[[Value]].
+          1. Let _result_ be _someValue_ if _hygienicTemp_ is *true*, and _someOtherValue_ otherwise.
         </emu-alg>
+
+        <p>Note that the resulting steps of ReturnIfAbrupt expansion may themselves contain ReturnIfAbrupt. In these cases, expansion can be applied to the resulting steps until the steps no longer contain ReturnIfAbrupt.</p>
       </emu-clause>
+
       <emu-clause id="sec-returnifabrupt-shorthands">
         <h1>ReturnIfAbrupt Shorthands</h1>
-        <p>Invocations of abstract operations and syntax-directed operations that are prefixed by `?` indicate that ReturnIfAbrupt should be applied to the resulting Completion Record. For example, the step:</p>
+        <p>Invocations of abstract operations may be prefixed by `?` as shorthand for applying the ReturnIfAbrupt expansion.</p>
         <emu-alg>
           1. ? OperationName().
         </emu-alg>
-        <p>is equivalent to the following step:</p>
+        <p>is equivalent to</p>
         <emu-alg>
           1. ReturnIfAbrupt(OperationName()).
         </emu-alg>
-        <p>Similarly, for method application style, the step:</p>
+        <p>wherever it appears. Similarly, prefix `?` may be used to apply the ReturnIfAbrupt expansion to the result of applying a syntax-directed operation.</p>
         <emu-alg>
-          1. ? _someValue_.OperationName().
+          1. Let _result_ be ? SyntaxDirectedOperation of |NonTerminal| with arguments _firstArgument_ and _secondArgument_.
         </emu-alg>
-        <p>is equivalent to:</p>
+        <p>is equivalent to the sequence of steps:</p>
         <emu-alg>
-          1. ReturnIfAbrupt(_someValue_.OperationName()).
+          1. Let _result_ be SyntaxDirectedOperation of |NonTerminal| with arguments _firstArgument_ and _secondArgument_.
+          1. ReturnIfAbrupt(_result_).
```

Comment:
This doesn't follow the given expansion. That would be
```
1. Let _result_ be ReturnIfAbrupt(SyntaxDirectedOperation of |NonTerminal| with arguments _firstArgument_ and _secondArgument_).
```

Reply:
GitHub seems to have lost the position of this, and I can't figure out what you're referring to. Is there an issue in one of my example expansions somewhere?

Reply:
github's position (line 859) is correct, but the example has changed. Updated, my point it that:
```
1. Let _result_ be ? SyntaxDirectedOperation of |NonTerminal|.
```
should (according to the definition) expand to:
```
1. Let _result_ be ReturnIfAbrupt(SyntaxDirectedOperation of |NonTerminal|).
```
as opposed to:
```
1. Let _result_ be SyntaxDirectedOperation of |NonTerminal|.
1. ReturnIfAbrupt(_result_).
```